internal/configs/virtualserver.go

```diff
@@ -490,169 +546,200 @@ type policiesCfg struct {
-				if pol.Spec.EgressMTLS.TrustedCertSecret != "" && !trustedEsists {
-					vsc.addWarningf(owner, `EgressMTLS policy %q references a Secret which does not exist`, key)
-					policyError = true
-					break
-				}
+	trustedCAFileName, trustedExists := egressMTLSSecrets[TrustedCertSecret]
+	if egressMTLS.TrustedCertSecret != "" && !trustedExists {
+		res.addWarning("EgressMTLS policy %q references a Secret which does not exist", polKey)
+		res.isError = true
+		return
+	}
 
-				egressMTLSPemFileName, tlsExists := egressMTLSSecrets[egressTLSSecret]
-				if pol.Spec.EgressMTLS.TLSSecret != "" && !tlsExists {
-					vsc.addWarningf(owner, `EgressMTLS policy %q references a Secret which does not exist`, key)
-					policyError = true
-					break
-				}
+	egressMTLSPemFileName, tlsExists := egressMTLSSecrets[egressTLSSecret]
+	if egressMTLS.TLSSecret != "" && !tlsExists {
+		res.addWarning("EgressMTLS policy %q references a Secret which does not exist", polKey)
+		res.isError = true
+		return
+	}
 
-				egressMTLS = &version2.EgressMTLS{
-					Certificate:    egressMTLSPemFileName,
-					CertificateKey: egressMTLSPemFileName,
-					Ciphers:        generateString(pol.Spec.EgressMTLS.Ciphers, "DEFAULT"),
-					Protocols:      generateString(pol.Spec.EgressMTLS.Protocols, "TLSv1 TLSv1.1 TLSv1.2"),
-					VerifyServer:   pol.Spec.EgressMTLS.VerifyServer,
-					VerifyDepth:    generateIntFromPointer(pol.Spec.EgressMTLS.VerifyDepth, 1),
-					SessionReuse:   generateBool(pol.Spec.EgressMTLS.SessionReuse, true),
-					ServerName:     pol.Spec.EgressMTLS.ServerName,
-					TrustedCert:    trustedCAFileName,
-					SSLName:        generateString(pol.Spec.EgressMTLS.SSLName, "$proxy_host"),
-				}
+	p.EgressMTLS = &version2.EgressMTLS{
+		Certificate:    egressMTLSPemFileName,
+		CertificateKey: egressMTLSPemFileName,
+		Ciphers:        generateString(egressMTLS.Ciphers, "DEFAULT"),
+		Protocols:      generateString(egressMTLS.Protocols, "TLSv1 TLSv1.1 TLSv1.2"),
+		VerifyServer:   egressMTLS.VerifyServer,
+		VerifyDepth:    generateIntFromPointer(egressMTLS.VerifyDepth, 1),
+		SessionReuse:   generateBool(egressMTLS.SessionReuse, true),
+		ServerName:     egressMTLS.ServerName,
+		TrustedCert:    trustedCAFileName,
+		SSLName:        generateString(egressMTLS.SSLName, "$proxy_host"),
+	}
+}
+
+func (vsc *virtualServerConfigurator) generatePolicies(ownerDetails policyOwnerDetails, policyRefs []conf_v1.PolicyReference, policies map[string]*conf_v1alpha1.Policy, context string, policyOpts policyOptions) policiesCfg {
+	config := newPoliciesConfig()
+
+	for _, p := range policyRefs {
+		polNamespace := p.Namespace
+		if polNamespace == "" {
+			polNamespace = ownerDetails.ownerNamespace
+		}
+
+		key := fmt.Sprintf("%s/%s", polNamespace, p.Name)
 
+		if pol, exists := policies[key]; exists {
+			res := newValidationResults()
+			switch {
+			case pol.Spec.AccessControl != nil:
+				config.addAccessControlConfig(res, pol.Spec.AccessControl)
```

Comment:
I wonder if we really need to pass `res` as a parameter. Could those add...Config functions return validationResults?

for example,
```go
func (p *policiesCfg) addAccessControlConfig(accessControl *conf_v1alpha1.AccessControl) validationResults
```
so that
```go

var res validationResults
. . .
res = config.addAccessControlConfig(pol.Spec.AccessControl)
```

imho it's easier to read when errors are returned this way.

Reply:
Looks much better, I agree. I think I went for this approach because I mistakenly thought the results were being built up over multiple iterations of the for loop, but they aren't.